### PR TITLE
Suggestions from Clippy

### PR DESCRIPTION
**Pull request by CGMossa**
While going through building wezterm, I found out there are a few complaints from clippy. These are the suggestions.

### CODE REVIEW

File: termwiz/src/terminal/windows.rs [hunk 1]
```diff
@@ -802,8 +801,8 @@ impl Terminal for WindowsTerminal {
                         "failed to WaitForMultipleObjects: {}",
                         IoError::last_os_error()
                     );
-                } else if result == WAIT_TIMEOUT {
-                    return Ok(None);
+                // } else if result == WAIT_TIMEOUT {
```
Comment (wez) [inline termwiz/src/terminal/windows.rs:804]: I don't agree with most of clippy's suggestions.  They're biased to ultimate minimalism, but they cause some information to be lost.  For example, in this case, it is very important for `WAIT_TIMEOUT` to be handled.
In the future, if some work is done here around other cases, the proposed change to implicitly handle it in the common case will be lost and there is risk of regression.

**Comment by CGMossa** [comment]
No problem.

You can close this or decline it as well.

The deep dive was worth it for me atleast.

File: wezterm-font/src/locator/mod.rs [hunk 2]
```diff
@@ -44,6 +44,12 @@ pub enum FontDataSource {
     },
 }
 
+impl std::hash::Hash for FontDataSource {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        core::mem::discriminant(self).hash(state);
```
Comment (wez) [inline wezterm-font/src/locator/mod.rs:49]: This feels like a net negative change: why remove the derived Hash in favor of whatever this is doing?

Comment (CGMossa) [reply wezterm-font/src/locator/mod.rs:49]: The the partial eq and hash has to match. Making both explicit should atleast make sure that we ignore fields equally, etc.
This should be equivalent to the derived result.

**Review by wez — CHANGES_REQUESTED** [review]
Thanks for taking the time to put this together.

However, I don't want most of these suggestions; they're generally not good for reasons that clippy doesn't understand.

The `write` -> `write_all` suggestions seem righteous, and I'll just fix those now to save you from coming back to this!

**Pull request closed without merging**